packages/ckeditor5-engine/src/view/placeholder.ts

```diff
@@ -50,24 +50,22 @@ export function enablePlaceholder( { view, element, text, isDirectHost = true, k
 } ): void {
 	const doc = view.document;
 
-	// Use a single a single post fixer per—document to update all placeholders.
+	// Use a single post fixer per—document to update all placeholders.
 	if ( !documentPlaceholders.has( doc ) ) {
 		documentPlaceholders.set( doc, new Map() );
 
 		// If a post-fixer callback makes a change, it should return `true` so other post–fixers
 		// can re–evaluate the document again.
-		doc.registerPostFixer( writer => updateDocumentPlaceholders( doc, writer ) );
+		doc.registerPostFixer( writer => updateDocumentPlaceholders( documentPlaceholders.get( doc )!, writer ) );
```

Comment:
Because I want to control the elements we iterate from outside. I use it [here](https://github.com/ckeditor/ckeditor5/blob/8099d921fe91369d9a9a79af13dbc5f5183dce98/packages/ckeditor5-engine/src/view/placeholder.ts#L93) to improve performance.

More info: https://github.com/ckeditor/ckeditor5/pull/17589#issuecomment-2535290791